include/drivers/misc/README

```diff
@@ -0,0 +1,17 @@
+Miscellaneous Drivers
```

Comment:
I think that this information should go somewhere down to `doc/reference`

Reply:
Proposed solution from API meeting recommended to place it here:

>Add a include/drivers/misc/README to clarify what goes in there. In that README, there should be a description of each item in the folder, with an explanation of why it doesn't conform to the standard API.

Reply:
ok, fair enough. Could it be `.rst` and have some minimal formatting then?

Reply:
The proposal was to put `README`. I'm not sure if omitting `.rst` or `.md` was intentional. @carlescufi, please comment